pandas/io/parsers/base_parser.py

```diff
@@ -679,14 +688,15 @@ def _set(x) -> int:
 
         return noconvert_columns
 
-    def _infer_types(self, values, na_values, try_num_bool: bool = True):
+    def _infer_types(self, values, na_values, cast_type, try_num_bool: bool = True):
         """
         Infer types of values, possibly casting
 
         Parameters
         ----------
         values : ndarray
         na_values : set
+        cast_type: Specifies if we want to cast explicitly
```

Comment:
Could we make this bool? Looks like we only need to check that it's no `None`?

Reply:
Changed